src/deepsparse/transformers/haystack/nodes.py

```diff
@@ -320,6 +307,25 @@ def load(*args, **kwargs):
             "DeepSparse Engine does not support loading from files"
         )
 
+    def _documents_to_passage_inputs(self, docs: List[Document]) -> List[str]:
+        # Optionally appends title
+        #
+        # :param docs: documents to turn into inputs
+        # :return: a list of raw text inputs
+        return [
+            f"{doc.meta['title']}{self.title_separator}{doc.content}"
+            if hasattr(doc, "meta")
+            and doc.meta.get("title", None) is not None
+            and self.title_separator is not None
+            else doc.content
```

Comment:
Ahh I specifically meant this inside of the loop here, and then in the other function do `passage_inputs = [self._document_to_passage(doc) for doc in docs]`

Then this becomes easy to unit test because you can call it with a single document and make sure the behavior matches expected
```python
doc = Document(meta=None, content="asdf") # or however you create documents
assert obj._document_to_passage(doc) == "asdf"

doc = Document(meta={"title": "qwer", content="asdf") # or however you create documents
assert obj._document_to_passage(doc) == "qwer asdf"
```

And it also makes it less dependendant on receiving a lists of documents as input. for example if you get a dictionary of documents or some other data structure, only the iteration of that structure has to change while the single document processing function (_document_to_passage) doesn't have to change at all

Reply:
Ah yeah, that's better. Making those changes...